TasksApp/TasksApp/ViewControllers/ContainerViewController.swift

```diff
@@ -0,0 +1,151 @@
+        menuViewController.delegate = self
+        
+        let navigationVC = UINavigationController(rootViewController: homeViewController)
+        addChild(navigationVC)
+        view.addSubview(navigationVC.view)
+        navigationVC.didMove(toParent: self)
+        self.navigationVC = navigationVC
+        homeViewController.delegate = self
+    }
+    
+    private func configureGestures() {
+        let swipeLeftGesture = UISwipeGestureRecognizer(target: self, action: #selector(toggleMenu))
+        swipeLeftGesture.direction = .left
+        swipeLeftGesture.cancelsTouchesInView = false
+        view.addGestureRecognizer(swipeLeftGesture)
+        
+        let swipeRightGesture = UISwipeGestureRecognizer(target: self, action: #selector(toggleMenu))
+        swipeRightGesture.direction = .right
+        swipeRightGesture.cancelsTouchesInView = false
+        view.addGestureRecognizer(swipeRightGesture)
+    }
+    
+    private func open(_ task: UIViewController) {
+        homeViewController.addChild(task)
+        homeViewController.view.addSubview(task.view)
+        task.view.frame = homeViewController.view.frame
+        task.didMove(toParent: homeViewController)
+        homeViewController.title = task.title
+        tasksControllers.append(task)
+    }
+    
+    private func resetToHome() {
+        tasksControllers.forEach { taskVC in
+            taskVC.view.removeFromSuperview()
+            taskVC.didMove(toParent: nil)
+        }
+        homeViewController.title = Constants.homeViewControllerTitle
+    }
+    
+    @objc private func toggleMenu() {
+        switch menuState {
+        case .closed:
+            UIView.animate(withDuration: 0.5,
+                           delay: 0, usingSpringWithDamping: 0.8,
+                           initialSpringVelocity: 0,
+                           options: .curveEaseInOut) {
+                
+                self.navigationVC?.view.frame.origin.x = self.homeViewController.view.frame.size.width - 100
+                
+            } completion: { [weak self] _ in
+                self?.menuState = .opened
+            }
+            
+        case .opened:
+            UIView.animate(withDuration: 0.5,
+                           delay: 0,
+                           usingSpringWithDamping: 0.8,
+                           initialSpringVelocity: 0,
+                           options: .curveEaseInOut) {
+                
+                self.navigationVC?.view.frame.origin.x = 0
+                
+            } completion: { [weak self] _ in
+                self?.menuState = .closed
+            }
+        }
+    }
+}
+
+// MARK: - HomeViewControllerDelegate
+
+extension ContainerViewController: HomeViewControllerDelegate {
+    
+    func didTappedButtonMenu() {
+        toggleMenu()
+    }
+}
+
+// MARK: - MenuViewControllerDelegate
+
+extension ContainerViewController: MenuViewControllerDelegate {
+    
+    func didSelect(menuItem: MenuViewController.MenuOptions) {
```

Comment:
should this be in past tense?

Reply:
I named this method in the past tense because it works after the user taps on a cell in the MenuViewController